lib/posthog/noop_worker.rb

```diff
@@ -0,0 +1,20 @@
+# A worker that doesn't consume jobs
+class PostHog
+  class NoopWorker
+    def initialize(queue)
+      @queue = queue
+    end
+
+    def run
+      # Does nothing
+    end
+
+    def flush
+      @queue.clear
```

Comment:
If the worker is `no-op` it feels like this should this be no-op too...

Maybe that pushes us to a different naming 🤔 

If we had `EventSendingWorker` and `EventSwallowingWorker` is that clearer?

Reply:
Interesting!

Look, I wrestled with this one and ultimately went the other way, but it's not a strongly held opinion.

What ultimately made me think flushing was OK is:
1. You can call `Client#dequeue_last_message` which pops the message off the queue
2. When writing tests it's not uncommon to reset state between tests. I am not personally doing this.
3. We can't write a flush test for the client in this project unless we actually clear the queue. We could potentially manipulate the queue directly in the test to achieve the desired effect though.

Reply:
Actually, I thought of another possibility. I definitely felt tension when I was adding an argument to `ensure_worker_running`.

What if clearing and flushing are being conflated? The client is the owner of the queue, not the worker, so what if we exposed a clear method on the client itself?

You could imagine doing the following:
``` ruby
client.flush # Blocks and waits until the queue is cleared
client.clear # Clears the queue without waiting
```

That would mean the `NoopWorker` could go back to being literally a noop, we could reach for `Client#clear` in the test, and consumers of the client, like my project, could reach for `Client#clear` in tests. 

It would also solve another problem I've been wrestling with which is how would I write a test for this....

``` ruby
def perform
  client.capture(...)
  client.flush
end
```

By the time I put an assertion the queue could be clear with the code as it is at the time of writing. If we changed it based on the above then `#flush` would noop, so I could do my assertion, and `#clear` would blow it away so I could reset state between tests.

That genuinely feels like a `NoopWorker` then. How does that sound to you?

Reply:
Not at all! This was a bit of a brain itch for me so I'm glad we persevered. I'm grateful for the feedback :smile:

Reply:
@pauldambra OK, I've made the change, rebased to squash the fixups, and pushed. I'm really happy with this API because it doesn't have the awkwardness we felt in the previous iteration.

The only caveat is that we couldn't get away from monkey patching the client in the `#flush` spec, but I think this is an acceptable trade off as it's completely isolated to that spec. I feel like this is cleaner than having `DummyWorker` in spec_helper.rb, but will ultimately support whatever position you have.

All feedback welcome.